content/blogs/2024-02-28-dbt-or-sqlmesh.md

```diff
@@ -0,0 +1,244 @@
+
+
+- SQLMesh comes with different user interfaces: a web UI and a CLI. Both are in the open-source version. 
+
+
+Comparing SQLMesh's exposed commands, like `sqlmesh plan`, and the way it interacts with the data warehouse, it evokes a strong resemblance to Terraform's approach. Adding UI on top is the realization that different user interfaces (Terraform being one) are important to support any user experiences.
+
+![sqlmesh ui](/blogs/2024-02-27-dbt-or-sqlmesh/sqlmesh-ui.gif)
+
+Here is a general feature comparison between dbt and SQLMesh:
+
+**Feature Comparison: dbt vs. SQLMesh**
+
+| Feature | dbt | SQLMesh |
+| --- | --- | --- |
+| **Modeling** | | |
+| SQL models | Yes | Yes |
+| Python models | Yes | Yes |
+| Jinja support | Yes | Yes |
+| Jinja macros | Yes | Yes |
+| Python macros | No | Yes |
+| **Validation** | | |
+| SQL semantic validation | No | Yes |
+| Unit tests | No | Yes |
+| Table diff | No | Yes |
+| Data audits | Yes | Yes |
+| Schema contracts | Yes | Yes |
+| Data contracts | No | Yes |
+| **Deployment** | | | 
+| Virtual Data Environments | No | Yes |
+| Open-source CI/CD bot | No | Yes |
+| Data consistency enforcement | No | Yes |
+| Native Airflow integration | No | Yes |
+| **Interfaces** | | |
+| CLI | Yes | Yes |
+| UI | Yes (Cloud) | Yes (OSS) |
+| Native Notebook Support | No | Yes |
+| Visualization | No | Yes |
+| Documentation generation | Yes | Yes |
+| Column-level lineage | Yes (Cloud) | Yes (OSS) |
+| **Miscellaneous** | | | 
+| Package manager | Yes | No |
+| Multi-repository support | No | Yes |
+| SQL transpilation | No | Yes |
+
+> **Note:** This table is for informational purposes only and may not reflect the latest features or capabilities of each tool. Please consult the official documentation for the most up-to-date information.
+
+
+
+## dbt Cloud vs. SQL Mesh commercial product
+
+dbt Core remains the foundation of the dbt open-source project, however, recent development efforts have primarily focused on the Cloud offering. This has coincided with both [pricing adjustments](https://www.getdbt.com/blog/consumption-based-pricing-and-the-future-of-dbt-cloud) and a perceived slowdown in core feature development. It's worth mentioning that dbt Cloud recently introduced column-level lineage (so only in their commercial product) built on top of [SQLMesh's open-source technology SQLGlot](https://github.com/tobymao/sqlglot)
+
+At Kestra we have seen many users moving away from dbt Cloud because of thos pricing changes and use Kestra with dbt Core to model their data and manage dependencies.
+
+On the other side, the commercial product associated with SQLMesh [is still under development](https://sqlmesh.readthedocs.io/en/stable/faq/faq/#company) and is expected to include features such as:
+
+- Model execution observability and monitoring tools
+- An enterprise-focused GitHub Actions CI/CD application
+- An advanced version of the open-source CI/CD bot
+- Hands-on support for companies using SQLMesh
```

Comment:
```suggestion
dbt Core remains the foundation of the dbt open-source project. However, recent development efforts have primarily focused on the Cloud offering. This has coincided with both [pricing adjustments](https://www.getdbt.com/blog/consumption-based-pricing-and-the-future-of-dbt-cloud) and a perceived slowdown in core feature development. It's worth mentioning that dbt Cloud recently introduced column-level lineage (so only in their commercial product) built on top of SQLMesh's open-source technology [SQLGlot](https://github.com/tobymao/sqlglot).

At Kestra, we have seen many users moving away from dbt Cloud because of those pricing changes and using Kestra with dbt Core to model their data and manage dependencies.

On the other side, the commercial product associated with SQLMesh is still [under development](https://sqlmesh.readthedocs.io/en/stable/faq/faq/#company) and is expected to include features such as:

- Model execution observability and monitoring tools
- An enterprise-focused GitHub Actions CI/CD application
- An advanced version of the open-source CI/CD bot
- Hands-on support for companies using SQLMesh
```